src/platform/packages/private/kbn-esql-editor/src/telemetry/events_registration.ts

```diff
@@ -66,4 +69,45 @@ export const registerESQLEditorAnalyticsEvents = once((analytics: AnalyticsServi
       },
     },
   });
+
+  analytics.registerEventType({
+    eventType: ESQL_QUERY_HISTORY_OPENED,
+    schema: {},
+  });
+  analytics.registerEventType({
+    eventType: ESQL_QUERY_HISTORY_CLICKED,
+    schema: {},
+  });
+  analytics.registerEventType({
+    eventType: ESQL_STARRED_QUERY_CLICKED,
+    schema: {},
+  });
+  analytics.registerEventType({
+    eventType: ESQL_QUERY_SUBMITTED,
+    schema: {
+      exec_source: {
+        type: 'keyword',
+        _meta: {
+          description:
+            'The source of the execution. Possible values are: manual|help|history|starred',
+        },
+      },
+      query_length: {
+        type: 'keyword',
+        _meta: { description: 'The length of the query.' },
+      },
+      query_lines: {
+        type: 'keyword',
+        _meta: { description: 'The number of lines in the query.' },
+      },
+      anti_limit_before_aggregate: {
+        type: 'boolean',
+        _meta: { description: 'Whether the query had a limit before the aggregate.' },
+      },
+      anti_missing_sort_before_limit: {
+        type: 'boolean',
+        _meta: { description: 'Whether the query was missing a sort before the limit.' },
```

Comment:
```suggestion
        _meta: { description: 'Whether the query was missing a sort before a limit.' },
```

This is a nit. Let's be more generic